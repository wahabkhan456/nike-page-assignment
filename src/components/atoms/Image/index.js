import React from "react";

export default function index({ image }) {
  return (
    <div>
      <img src={image} alt="Shoe" />
    </div>
  );
}
