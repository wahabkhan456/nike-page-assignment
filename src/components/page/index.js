import Selection from "./Selection/Selection";

export { Selection };
