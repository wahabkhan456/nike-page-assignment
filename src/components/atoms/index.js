import NavItem from "./nav-item/NavItem";
import Image from "../atoms/Image";
import Size from "./Size/Size";
import Color from "./Color/Color";

export { NavItem, Image, Size, Color };
