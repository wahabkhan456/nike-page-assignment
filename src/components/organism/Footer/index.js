import React from "react";
import { NavItem } from "../../atoms/index";
import "./Footer.css";
import { useDispatch } from "react-redux";
import { addToCart } from "../../../redux/actions";
import alertify from "alertifyjs";
import { useTranslation } from "react-i18next";

export default function Index() {
  const dispatch = useDispatch();
  const [t] = useTranslation();

  return (
    <div className="Footer">
      <div className="prev-next">
        <div>
          <h5 className="-button">{t("prev")}</h5>
          <NavItem class="icon large">
            <i class="fas fa-long-arrow-alt-left"></i>
          </NavItem>
        </div>
        <div>
          <h5 className="-button">{t("next")}</h5>
          <NavItem class="icon large">
            <i class="fas fa-long-arrow-alt-right"></i>
          </NavItem>
        </div>
      </div>
      <div
        className="price"
        onClick={() => {
          dispatch(addToCart());
          alertify.success(t("success"));
        }}
      >
        <NavItem class="large nav-link">{t("add_cart")} 9.8$</NavItem>
      </div>
    </div>
  );
}
