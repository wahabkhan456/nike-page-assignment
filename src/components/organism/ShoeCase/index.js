import React, { useState } from "react";
import "./shoecase.css";
import { Image } from "../../atoms";
import { useSelector } from "react-redux";

export default function Index() {
  const allImages = useSelector((state) => state.getImages.images);
  console.log(allImages);
  const [current, setCurrent] = useState(0);

  const getNext = () => {
    if (current === 2) {
      setCurrent(0);
    } else {
      setCurrent(current + 1);
    }
  };

  const getPrev = () => {
    if (current === 0) {
      setCurrent(2);
    } else {
      setCurrent(current - 1);
    }
  };

  return (
    <div className="show-case">
      <div className="shoe-image">
        <Image image={allImages[current]} />
      </div>
      <div className="rotation"></div>
      <div className="rotation-buttons">
        <i class="fa fa-caret-left icon" onClick={() => getPrev()}></i>
        <i class="fa fa-caret-right icon" onClick={() => getNext()}></i>
      </div>
    </div>
  );
}
