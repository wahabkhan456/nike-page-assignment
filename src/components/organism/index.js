import Header from "./Header/Header";
import Footer from "./Footer";
import ShoeCase from "./ShoeCase";
import ShoeInfo from "./ShoeInfo/ShoeInfo";
import SizeSelection from "./SizeSelection/SizeSelection";

export { Header, Footer, ShoeCase, ShoeInfo, SizeSelection };
