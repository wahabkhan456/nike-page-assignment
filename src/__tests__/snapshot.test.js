import React from "react";
import renderer from "react-test-renderer";
import { ShoeInfo, SizeSelection } from "../components/organism";
import { Size, Image } from "../components/atoms";

it("Show Info Component", () => {
  const tree = renderer.create(<ShoeInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Size Selection Component", () => {
  const tree = renderer.create(<SizeSelection />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Size Component", () => {
  const tree = renderer.create(<Size />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Image Component", () => {
  const tree = renderer.create(<Image />).toJSON();
  expect(tree).toMatchSnapshot();
});
