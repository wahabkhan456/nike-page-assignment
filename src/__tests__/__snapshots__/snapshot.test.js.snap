// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component 1`] = `
<div>
  <img
    alt="Shoe"
  />
</div>
`;

exports[`Show Info Component 1`] = `
<div>
  <h5
    className="catagory"
  >
    catagory
  </h5>
  <h5
    className="shoe-name"
  >
    name
  </h5>
  <p
    className="shoe-info"
  >
    shoe_info
  </p>
  <div
    className="video-container"
  >
    <div
      className="Video"
    >
      <i
        class="fa fa-caret-right"
      />
    </div>
    <h5
      className="play-video"
    >
      video
    </h5>
  </div>
</div>
`;

exports[`Size Component 1`] = `
<div
  className="size-container"
>
  <h6
    className="size"
  />
</div>
`;

exports[`Size Selection Component 1`] = `
<div>
  <h5
    className="size-selection-title"
  >
    s_s
  </h5>
  <div
    className="Size-container"
  >
    <div
      className="size-container"
    >
      <h6
        className="size"
      >
        8
      </h6>
    </div>
    <div
      className="size-container"
    >
      <h6
        className="size"
      >
        8.5
      </h6>
    </div>
    <div
      className="size-container"
    >
      <h6
        className="size"
      >
        9
      </h6>
    </div>
    <div
      className="size-container"
    >
      <h6
        className="size"
      >
        9.5
      </h6>
    </div>
    <div
      className="size-container"
    >
      <h6
        className="size"
      >
        10
      </h6>
    </div>
  </div>
  <div
    className="Size-container"
  >
    <div
      className="size-container"
    >
      <h6
        className="size"
      >
        10.5
      </h6>
    </div>
    <div
      className="size-container"
    >
      <h6
        className="size"
      >
        11
      </h6>
    </div>
    <div
      className="size-container"
    >
      <h6
        className="size"
      >
        11.5
      </h6>
    </div>
    <div
      className="size-container"
    >
      <h6
        className="size"
      >
        12
      </h6>
    </div>
    <div
      className="size-container"
    >
      <h6
        className="size"
      >
        12.5
      </h6>
    </div>
  </div>
</div>
`;
